pywinauto/controls/uia_controls.py

```diff
@@ -575,17 +605,33 @@ def item_count(self):
             # TODO: This could be implemented by getting custom ItemCount Property using RegisterProperty
             # TODO: See https://msdn.microsoft.com/ru-ru/library/windows/desktop/ff486373%28v=vs.85%29.aspx for details
             # TODO: comtypes doesn't seem to support IUIAutomationRegistrar interface
-            return (len(self.children()))
+            return self.__resolve_row_count(len(self.children()))
 
     # -----------------------------------------------------------
     def column_count(self):
         """Return the number of columns"""
         if self.iface_grid_support:
             return self.iface_grid.CurrentColumnCount
+        elif self.is_table and len(self.children()) > 0:
+            return self.__resolve_col_count(len(self.children()[0].children()))
         else:
             # ListBox doesn't have columns
             return 0
 
+    # -----------------------------------------------------------
+    def get_column_header_items(self):
+        """Return Headers of each column (top of DataGrid)"""
+        if self.col_header:
+            return get_item(0).children(control_type="Header")
```

Comment:
Must be `self.get_item(...)` and `self.get_column(...)` below (caught by Codacy as well).